### DISCUSSION TITLE
mypy validation not working when including pydantic class from custom library

### DISCUSSION THREAD

**Discussion by khushbuadav** [pydantic/pydantic#3225] [Question]
We have different python services which share common schema(`pydantic`) types which are defined in a custom python library. This package is installed system-wide (`pip install`) and then other services  just do `from companylib import schemas`
We are using mypy for validating python object creations and hoping to catch obvious errors. However the plugin does not raise any validation error when called from system installed library. If we move the `schemas` files to local folder, it works.

companylib/schemas.py
```python
from pydantic import BaseModel

class VFrame(BaseModel):
    id: Optional[int] = None
    txn_id: int
    name: str
    uri: str
    props: str

    class Config:
        orm_mode = True
```
service1/app.py
``` python
from companylib.schemas import VFrame

frame_in = VFrame(txn_id=5, name="frame-1",  uri="s3://frame-1") # <== No error here for props
```

--------

When we move the schemas.py to service1 directory, it works.
```
service1/
    app.py
    schemas.py
```

service1/app.py
``` python
from .schemas import VFrame

frame_in = VFrame(txn_id=5, name="frame-1",  uri="s3://frame-1") # <== error here for props
```

However, this would mean we replicate the companylib to all our services, which is not a solution.
Thanks in advance!

**Comment by daviskirk** [comment]
Depending on how you install and package "companylib" mypy might not be registering the types at all.
The package that provides companylib needs to have a "py.typed" module included in the package (see https://www.python.org/dev/peps/pep-0561/#packaging-type-information) for mypy to know that your package provides type information (for example, pydantic itself has this here: https://github.com/samuelcolvin/pydantic/blob/master/pydantic/py.typed).

So, when using it in your local package, it's importing "BaseModel" directly from pydantic and mypy understands that it's typed (because pydantic provided the `py.typed` marker.

But when your importing the model from `companylib` your just importing a random variable from a python library that has no such marker, so mypy doesn't attempt to pick it up.

At least that COULD be the problem you're having.

**Reply by khushbuadav** [reply]
Thanks @daviskirk for the reply. Even we think that is the issue.
But since this is a very basic use case (sharing a common code as lib and validating / autocompleting against it), am surprised there is no standard solution to it. Or if there is its not easily available to general public.
So in our case, do we bundle a `.typed` file with the repo?
